src/game/objects/general/door.c

```diff
@@ -119,7 +119,7 @@ void Door_Initialise(int16_t item_num)
     z_sector = ((item->pos.z - r->z) >> WALL_SHIFT) + dx;
     x_sector = ((item->pos.x - r->x) >> WALL_SHIFT) + dy;
     door->d1.sector = &r->sectors[z_sector + x_sector * r->z_size];
-    room_num = Room_GetDoor(door->d1.sector);
+    room_num = door->d1.sector->portal_room.wall;
```

Comment:
Below it looks like there is a lot of repeated code like:
```
        b = &g_RoomInfo[room_num];
        z_sector = ((item->pos.z - b->z) >> WALL_SHIFT) + dx;
        x_sector = ((item->pos.x - b->x) >> WALL_SHIFT) + dy;
        box_num = b->sectors[z_sector + x_sector * b->z_size].box;
```

Is it worth another function in `room.c` similar to `Room_GetPitSector`. Something like `Room_GetBoxNumber`?

Reply:
Yeah, there are some cases like this left. What would be ideal in `C` is lambda functions so we could iterate commonly and "do the thing" per-case.